CMakeLists.txt

```diff
@@ -256,6 +256,53 @@ include(cmake/ssl.cmake)
 include(cmake/upb.cmake)
 include(cmake/xxhash.cmake)
 include(cmake/zlib.cmake)
+include(cmake/xds_proto.cmake)
+
+# Setup external proto libraray at third_party/envoy-api
+if (NOT EXISTS ${CMAKE_CURRENT_SOURCE_DIR}/third_party/envoy-api)
+  fetch_check_extract(
+    ${CMAKE_CURRENT_SOURCE_DIR}/third_party/envoy-api
+    https://storage.googleapis.com/grpc-bazel-mirror/github.com/envoyproxy/data-plane-api/archive/18b54850c9b7ba29a4ab67cbd7ed7eab7b0bbdb2.tar.gz
```

Comment:
In bazel we actually provide 2 URLs  (the mirror and the default github one) and that's not reflected here, which can make the builds flaky.
Note that AFAIK users in china cannot access https://storage.googleapis.com/ (it's a google URL) so if you only give them this one URL, the build will be broken for them.